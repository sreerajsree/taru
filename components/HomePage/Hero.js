import { motion } from 'framer-motion';
import useIsMobile from '../../hooks/useIsMobile';

const paragraphAnimation = {
  hidden: { opacity: 1, scale: 0 },
  visible: {
    opacity: 1,
    scale: 1,
    transition: {
      delay: 0.3,
      when: 'beforeChildren',
      staggerChildren: 0.2,
    },
  },
};

const spanAnimation = {
  hidden: { y: 20, opacity: 0 },
  visible: {
    y: 0,
    opacity: 1,
  },
};

const renderTagline = (tagline) => {
  const titles = [];
  tagline.forEach((text, index) => {
    titles.push({ id: index, text });
    if (index + 1 < tagline.length) titles.push({ id: index, span: true });
  });

  return titles.map(({ id, text, span }) => {
    if (span) {
      return <motion.span key={`subtitle-span-${id}`} variants={spanAnimation} className="rounded-full inline-block h-2 w-2 mx-2 mt-1 bg-black" />;
    }
    return <motion.span key={`subtitle-${id}`} variants={spanAnimation}>{text}</motion.span>;
  });
};

export default function Hero({ title, tagline, backgroundImage }) {
  const isMobile = useIsMobile();
  const desktopImage = require(`../../images/home/${backgroundImage}?resize&size=2000`);
  const mobileImage = require(`../../images/home/${backgroundImage}?resize&size=1000`);
  const image = isMobile ? mobileImage : desktopImage;
  return (
    <section id="hero">
      <div
        className="w-screen h-screen md:bg-fixed flex justify-center items-center flex-col font-title bg-cover bg-top"
        style={{
          backgroundImage: `url('${image}')`,
        }}
      >
        <h1 className="text-center text-7xl block">{title}</h1>
        <motion.p
          variants={paragraphAnimation}
          initial="hidden"
          animate="visible"
          className="block mt-5 mb-20 flex items-center text-xl"
        >
          {renderTagline(tagline)}
        </motion.p>
      </div>
    </section>
  );
}
