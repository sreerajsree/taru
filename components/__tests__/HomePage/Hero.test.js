import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Hero from '../../HomePage/Hero';

jest.mock('react-use');

describe('Hero', () => {
  test('renders and matches snapshot', () => {
    const { asFragment } = render(<Hero title="Ameira Yanni" tagline={['copywriter', 'actor', 'creative']} backgroundImage="header-background.jpg" />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Ameira Yanni is rendered', () => {
    render(<Hero title="Ameira Yanni" tagline={['copywriter', 'actor', 'creative']} backgroundImage="header-background.jpg" />);
    expect(screen.getByText('Ameira Yanni')).toBeInTheDocument();
  });
});
