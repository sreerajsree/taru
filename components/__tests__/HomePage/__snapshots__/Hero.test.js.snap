// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders and matches snapshot 1`] = `
<DocumentFragment>
  <section
    id="hero"
  >
    <div
      class="w-screen h-screen md:bg-fixed flex justify-center items-center flex-col font-title bg-cover bg-top"
      style="background-image: url(test-file-stub);"
    >
      <h1
        class="text-center text-7xl block"
      >
        Ameira Yanni
      </h1>
      <p
        class="block mt-5 mb-20 flex items-center text-xl"
        style="opacity: 1; transform: scale(0) translateZ(0);"
      >
        <span
          style="opacity: 0; transform: translateY(20px) translateZ(0);"
        >
          copywriter
        </span>
        <span
          class="rounded-full inline-block h-2 w-2 mx-2 mt-1 bg-black"
          style="opacity: 0; transform: translateY(20px) translateZ(0);"
        />
        <span
          style="opacity: 0; transform: translateY(20px) translateZ(0);"
        >
          actor
        </span>
        <span
          class="rounded-full inline-block h-2 w-2 mx-2 mt-1 bg-black"
          style="opacity: 0; transform: translateY(20px) translateZ(0);"
        />
        <span
          style="opacity: 0; transform: translateY(20px) translateZ(0);"
        >
          creative
        </span>
      </p>
    </div>
  </section>
</DocumentFragment>
`;
